src/csharp/Grpc.Core/Internal/CallSafeHandle.cs

```diff
@@ -127,16 +128,52 @@ public void StartSendCloseFromClient(ISendCompletionCallback callback)
             }
         }
 
-        public void StartSendStatusFromServer(ISendStatusFromServerCompletionCallback callback, Status status, MetadataArraySafeHandle metadataArray, bool sendEmptyInitialMetadata,
+        public unsafe void StartSendStatusFromServer(ISendStatusFromServerCompletionCallback callback, Status status, MetadataArraySafeHandle metadataArray, bool sendEmptyInitialMetadata,
```

Comment:
let's not change the signature of this method but use unsafe block in the body (the signature is generated and it's not worth the trouble changing the template).

Reply:
adding `unsafe` doesn't change the signature; if you'd rather I used `unsafe` internally, that's fine - I just want to make sure that we're doing it for the right reasons